platform/logging/__tests__/LoggingService.test.ts

```diff
@@ -0,0 +1,74 @@
+import { BehaviorSubject } from 'rxjs';
+import { LogLevelId } from '../LogLevel';
+import { MutableLoggerFactory } from '../LoggerFactory';
+import { LoggingConfig } from '../LoggingConfig';
+import { LoggingService } from '../LoggingService';
+
+const createRandomConfig = (appenderKey: string) => {
+  return new LoggingConfig({
+    appenders: new Map([
+      [
+        appenderKey,
+        { kind: 'console', layout: { kind: 'pattern', highlight: false } }
+      ]
+    ]),
+    loggers: new Map([
+      ['random', { appenders: [appenderKey], level: 'info' as LogLevelId }]
+    ])
+  });
+};
+
+const getLastMockCallArgs = (mockFunction: jest.Mock<Function>) => {
+  expect(mockFunction).toHaveBeenCalled();
+  return mockFunction.mock.calls[mockFunction.mock.calls.length - 1];
+};
+
+let factory: MutableLoggerFactory;
+let service: LoggingService;
+let updateConfigMock: jest.Mock<Function>;
+let closeMock: jest.Mock<Function>;
+beforeEach(() => {
+  factory = new MutableLoggerFactory();
+  updateConfigMock = jest
+    .spyOn(factory, 'updateConfig')
+    .mockImplementation(() => {});
+  closeMock = jest.spyOn(factory, 'close').mockImplementation(() => {});
+
+  service = new LoggingService(factory);
+});
+
+test('`upgrade()` updates logging factory config.', () => {
+  expect(factory.updateConfig).not.toHaveBeenCalled();
+
+  const config = createRandomConfig('old');
+  const config$ = new BehaviorSubject<LoggingConfig>(config);
+
+  service.upgrade(config$);
+
+  expect(updateConfigMock).toHaveBeenCalledTimes(1);
+  expect(getLastMockCallArgs(updateConfigMock)[0]).toBe(config);
```

Comment:
if I'm not mistaken `toHaveBeenLastCalledWith(..)` will succeed with _any_ instance of `LoggingConfig` (any combination of appenders and loggers). But I want to make sure that it's called exactly with what `config$` provides.

Am I missing something?